File: 2023/Primeira Fase/Estoque/estoque.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    int n, m, estoque[505][505];

    cin >> n >> m;

    for(int i=1; i<=n; i++){
        for(int j=1; j<=m; j++){
            cin >> estoque[i][j];
        }
    }
    
    int p;
    cin >> p;

    int resp=0;

    for(int i=0; i<p; i++){
        int a, b;
        cin >> a >> b;
        if(estoque[a][b]>0){
            estoque[a][b]--;
            resp++;
        }
    }
    cout << resp << endl;
    return 0;
}

File: 2023/Primeira Fase/Estoque/estoque.py
n, m = [int(x) for x in input().split()]

estoque = []

for i in range(n):
    linha = [int(x) for x in input().split()]
    estoque.append(linha)

p = int(input())

resp = 0

for i in range(p):
    a, b = [int(x) for x in input().split()]

    a-=1
    b-=1

    if estoque[a][b]>0:
        estoque[a][b]-=1
        resp+=1

print(resp)
